source/codegen/validate_examples.py

```diff
@@ -62,7 +62,7 @@ def _validate_examples(
         _system("poetry new .")
         _system("poetry add grpcio")
         _system("poetry add --dev grpcio-tools mypy mypy-protobuf types-protobuf grpc-stubs")
-        _system("poetry add --dev black")
+        _system("poetry add --dev black==23.3.0")
```

Comment:
Instead of locking down Black, why not reformat the code to use Black's 2024 style?

Cc: @mshafer-NI

Reply:
Considering this is halting builds, how much code would need to be changed? Perhaps both the work to update grpc-device's dependency versioning to using a poetry.lock file and pulling ni-python-styleguide instead of black can be completed in a future tech debt item, so that our current builds can succeed?

Reply:
5 files

```
(.venv) PS D:\dev\grpc-device> black .\examples\
reformatted D:\dev\grpc-device\examples\nirfmxdemod\ADemod-AM-basic.py
reformatted D:\dev\grpc-device\examples\nirfmxtdscdma\acp.py
reformatted D:\dev\grpc-device\examples\nirfmxtdscdma\modacc.py

All done! ✨ 🍰 ✨
3 files reformatted, 60 files left unchanged.
(.venv) PS D:\dev\grpc-device> black .\source\
reformatted D:\dev\grpc-device\source\codegen\format_mi_metadata.py
reformatted D:\dev\grpc-device\source\codegen\metadata_validation.py

All done! ✨ 🍰 ✨
2 files reformatted, 14 files left unchanged.
(.venv) PS D:\dev\grpc-device> git diff
diff --git a/examples/nirfmxdemod/ADemod-AM-basic.py b/examples/nirfmxdemod/ADemod-AM-basic.py
index 9e6fae29..7f5d4b58 100644
--- a/examples/nirfmxdemod/ADemod-AM-basic.py
+++ b/examples/nirfmxdemod/ADemod-AM-basic.py
@@ -32,7 +32,6 @@ If they are not passed in as command line arguments, then by default the server
 "localhost:31763", with "SimulatedDevice" as the resource name
 """

-
 import sys

 import grpc
diff --git a/examples/nirfmxtdscdma/acp.py b/examples/nirfmxtdscdma/acp.py
index 48f85756..6a9540d8 100644
--- a/examples/nirfmxtdscdma/acp.py
+++ b/examples/nirfmxtdscdma/acp.py
@@ -40,7 +40,6 @@ If they are not passed in as command line arguments, then by default the server
 "localhost:31763", with "SimulatedDevice" as the resource name
 """

-
 import sys

 import grpc
diff --git a/examples/nirfmxtdscdma/modacc.py b/examples/nirfmxtdscdma/modacc.py
index 452c79cc..11d8f97e 100644
--- a/examples/nirfmxtdscdma/modacc.py
+++ b/examples/nirfmxtdscdma/modacc.py
@@ -37,7 +37,6 @@ If they are not passed in as command line arguments, then by default the server
 "localhost:31763", with "SimulatedDevice" as the resource name
 """

-
 import sys

 import grpc
diff --git a/source/codegen/format_mi_metadata.py b/source/codegen/format_mi_metadata.py
index 93538040..6d9bf828 100644
--- a/source/codegen/format_mi_metadata.py
+++ b/source/codegen/format_mi_metadata.py
@@ -1,4 +1,5 @@
 """Script for formatting mi-driver metadata in same style as hapigen output."""
+
 import argparse
 import io
 import os
diff --git a/source/codegen/metadata_validation.py b/source/codegen/metadata_validation.py
index 471f8ea1..38c3d99e 100644
--- a/source/codegen/metadata_validation.py
+++ b/source/codegen/metadata_validation.py
@@ -1,4 +1,5 @@
 """Metadata validation."""
+
 from typing import Any, Dict, List, Set

 import common_helpers
```

Reply:
Deferring as tech debt is fine

Reply:
CC @maxxboehme